src/main/kotlin/io/csbroker/apiserver/repository/common/RedisRepository.kt

```diff
@@ -39,64 +39,43 @@ class RedisRepository(
         redisTemplate.delete(code)
     }
 
+    @Suppress("UNCHECKED_CAST")
     fun setRank(scoreMap: Map<String, Double>) {
-        redisTemplate.opsForZSet().add(
-            RANKING,
-            scoreMap.map { TypedTuple.of(it.key, it.value) }.toSet(),
+        redisTemplate.execute(
+            object : SessionCallback<Unit> {
+                override fun <K : Any?, V : Any?> execute(operations: RedisOperations<K, V>) {
+                    val stringOperations = operations as RedisOperations<String, String>
+                    stringOperations.multi()
+                    stringOperations.delete(RANKING)
+                    stringOperations.opsForZSet().add(
+                        RANKING,
+                        scoreMap.map { TypedTuple.of(it.key, it.value) }.toSet(),
+                    )
+                    stringOperations.exec()
+                }
+            },
         )
     }
 
     fun getRank(key: String): RankResultDto {
-        var rank: Long? = null
-
         val score = redisTemplate.opsForZSet().score(RANKING, key) ?: 0.0
```

Comment:
```suggestion
        val score = redisTemplate.opsForZSet().score(RANKING, key) ?: return RankResultDto(null)
```
score가 존재하지 않는다면 해당 키가 없다는거고 아래 연산은 더이상 의미가 없으니 early return 해줘도 좋을 것 같아요~